piston/resource.py

```diff
@@ -244,15 +244,14 @@ def _regex_domain_match(self, origin):
                 return origin
 
     def options(self):
-        obj = self._get_obj_or_none()
-        allowed_methods = ', '.join((method.upper() for method in self.get_allowed_methods(obj)))
-
-        http_headers = {'Allowed': allowed_methods}
-        if getattr(settings, 'PISTON_CORS', False):
-            http_headers.update({
-                ACCESS_CONTROL_ALLOW_METHODS: allowed_methods,
-                ACCESS_CONTROL_ALLOW_HEADERS: ', '.join(self._get_cors_allowed_headers()),
-            })
+        if getattr(settings, 'PISTON_CORS', False) and self.request.META.get('HTTP_ORIGIN'):
```

Comment:
to se mi nelíbí, ještě bych tu dal, and not request.user, nechci zničit options pro přihlášené uživatele